src/ComunicaEngine.js

```diff
@@ -22,23 +23,38 @@ export default class ComunicaEngine {
    * Creates an asynchronous iterable of results for the given SPARQL query.
    */
   async* execute(sparql, source) {
-    if ((/^\s*(?:INSERT|DELETE)/i).test(sparql))
+    if ((/^\s*(?:INSERT|DELETE)/i).test(sparql)) {
       yield* this.executeUpdate(sparql, source);
-
-    // Load the sources if passed, the default sources otherwise
-    const sources = await (source ? this.parseSources(source) : this._sources);
-    if (sources.length !== 0) {
-      // Execute the query and yield the results
-      const queryResult = await this._engine.query(sparql, { sources, ...this._options });
-      yield* this.streamToAsyncIterable(queryResult.bindingsStream);
```

Comment:
So the original code would return results here, and that was on purpose., I thought. Or was this a bug?

Reply:
As mentioned in the above comment - this wouldn't happen due to the fact that previously there was an error thrown by `executeUpdate` causing the iterator to stop iterating/yielding

Reply:
Haha, right. My bad.